Code/Tools/TestImpactFramework/Runtime/Native/Code/Source/Target/Native/TestImpactNativeTargetListCompiler.cpp

```diff
@@ -25,18 +25,32 @@ namespace TestImpact
         AZStd::vector<NativeProductionTarget> productionTargets;
         AZStd::vector<NativeTestTarget> testTargets;
 
+        size_t numDiscardedTestTargets = 0;
         for (auto&& descriptor : buildTargetDescriptors)
         {
-            // If this build target has an associated test artifact then it is a test target, otherwise it is a production target
             if (auto&& testTargetMeta = testTargetMetaMap.find(descriptor.m_name); testTargetMeta != testTargetMetaMap.end())
             {
                 AZ_TestImpact_Eval(descriptor.m_type == TargetType::TestTarget, ArtifactException, "Target has associated target meta but is a production target");
                 testTargets.emplace_back(AZStd::move(descriptor), AZStd::move(testTargetMeta->second));
             }
-            else
+            else if (descriptor.m_type == TargetType::ProductionTarget)
             {
                 productionTargets.emplace_back(AZStd::move(descriptor));
             }
+            else
+            {
+                // Test targets without an associated test target meta artifact are either not opted in to TIAF or not in the
+                // suite(s) to be run so will be skipped
+                numDiscardedTestTargets++;
+            }
+        }
+
+        if (numDiscardedTestTargets)
+        {
+            AZ_Info(
+                "CompileNativeTargetLists",
+                "%zu test targets were either opted-out of TIAF or not in the suite(s) selected for this run.\n",
+                numDiscardedTestTargets);
```

Comment:
Would it be better instead to append the test targets names without an associated test target meta artifacts in a string that can be printed out using AZ_Info statement.

It looks like the TestImpactTargetDescriptor.h has a name value which can be logged to notify the user of the CMake targets in question.

Reply:
Probably just do an AZ_Warn on after `numDiscardedTestTargets++` instead of a concatenated string in conjuction with this message.

Reply:
I'll have an experiment and follow this up with the PR that will contain the amendments for the stack trace on crash 👍